### PR TITLE
(v2) Feature: Implements Passport CAS Strategy

### PR DESCRIPTION
**Pull request by joker-x**
Use https://www.npmjs.com/package/passport-cas2

### CODE REVIEW

File: server/config/default.yaml [hunk 1]
```diff
@@ -171,6 +171,18 @@ postfixBounce:
...
 
+cas:
+  # enable to use CAS user backend
+  enabled: false
+  url: https://example.cas-server.com
+  # nameTag identifies the attribute to be used for user's full name
+  nameTag: username
+  # mailTag identifies the attribute to be used for user's email address
+  mailTag: mail
+  newUserRole: campaignsAdmin
```
Comment (bures) [inline server/config/default.yaml]: Could we have the newUserRole also set to nobody as in the docker entrypoint?

File: client/src/lib/page.js [hunk 2]
```diff
@@ -414,6 +414,11 @@ export class SectionContent extends Component {
     }
 
     componentDidMount() {
+        const queryParams = this.props.location.search;
+        if (queryParams.indexOf('cas-login-success') > -1) this.setFlashMessage('success', 'Successful authentication');
+        if (queryParams.indexOf('cas-logout-success') > -1) this.setFlashMessage('success', 'Successful logout');
+        if (queryParams.indexOf('cas-login-error') > -1) this.setFlashMessage('danger', 'Fail authentication');
```
Comment (bures) [inline client/src/lib/page.js]: "Authentication successful"
"Logout successful"
"Authentication failed"

File: client/src/lib/page.js [hunk 3]
```diff
@@ -602,7 +611,7 @@ export class DropdownLink extends Component {
 
         const clsName = "dropdown-item" + (props.className ? " " + props.className : "")
         return (
-            <Link to={props.to} className={clsName}>{props.children}</Link>
+            <Link to={props.to} className={clsName} onClick={() => window.location.href=props.to}>{props.children}</Link>
```
Comment (bures) [inline client/src/lib/page.js:617]: Why is this? I guess, this would make any DropdownLink refresh the client.

File: client/src/login/Login.js [hunk 4]
```diff
@@ -126,6 +126,25 @@ export default class Login extends Component {
                     </ButtonRow>
                 </Form>
             </div>
-        );
+          );
+        } else {
+          if (mailtrainConfig.isAuthenticated) {
+            return (
```
Comment (bures) [inline client/src/login/Login.js]: Why is there a logout on the login page? (And only for CAS?)

**Review by bures — CHANGES_REQUESTED** [review]
Looks good. Could you please take a look at the comments I left in the code? Thanks.

**Comment by bures** [comment]
I guess this is specific to Apereo CAS, isn't it? If one wanted to use for instance Keycloak, we would need another provider, ideally OAUTH2, is it? Thanks.

File: server/config/default.yaml (hunk 1 above)
Comment (joker-x) [reply server/config/default.yaml]: Ready https://github.com/Mailtrain-org/mailtrain/pull/960/commits/e980e4d34fe51c864d192599eda5acc5c71831e9

File: client/src/lib/page.js (hunk 3 above)
Comment (joker-x) [reply client/src/lib/page.js:617]: Ready https://github.com/Mailtrain-org/mailtrain/pull/960/commits/747d9f0c5143cb151abe26614feadeebb516104d

File: client/src/login/Login.js (hunk 4 above)
Comment (joker-x) [reply client/src/login/Login.js]: Ready https://github.com/Mailtrain-org/mailtrain/pull/960/commits/99e3f12c24e6392d9646e4141df03d382d74ca61

File: client/src/lib/page.js (hunk 2 above)
Comment (joker-x) [reply client/src/lib/page.js]: Ready https://github.com/Mailtrain-org/mailtrain/pull/960/commits/b8088acfaec80d6b37df0233d40b006c6501f3c3

**Comment by joker-x** [comment]
How docker resolve the installation of the package passport-cas2 if WITH_CAS is true?

**Review by bures — APPROVED** [review]

**Comment by bures** [comment]
Looks good. Sorry for my inactivity. I'll be happy to merge it if you are still interested. If so, would you please just check the conflicts that arose in the meantime? I'll merge it then. Thanks.

**Comment by bures** [comment]
Thanks.

**Pull request merged**